library/alloc/src/vec/in_place_collect.rs

```diff
@@ -191,14 +194,19 @@ where
             );
         }
 
-        // Drop any remaining values at the tail of the source but prevent drop of the allocation
-        // itself once IntoIter goes out of scope.
-        // If the drop panics then we also leak any elements collected into dst_buf.
+        // `forget_allocation_drop_remaining` drops any remaining values at the tail of the source
+        // but prevent drop of the allocation itself once `IntoIter` goes out of scope.
+        // `dst_guard` takes care of dropping the allocation and the new values if and only if
+        // `forget_allocation_drop_remaining` panics while dropping any of the old values.
+        // This is safe because `forget_allocation_drop_remaining` immediately forgets the allocation
+        // and won't panic before that.
```

Comment:
This section could be shortened by saying that allocation ownership is transferred to `InPlaceDstBufDrop` before any panic can happen and then point to the docs of `forget_allocation_drop_remaining`.

Reply:
I tried rewording the comment with ownership in mind. I'm not sure how I could have pointed to `forget_allocation_drop_remaining`'s docs while still shortening the comment.